### PR TITLE
Adding a map BG and explosions looks nicer

### PR DESCRIPTION
**Pull request by Octopus773**
## Information
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Non-breaking changes

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation

### CODE REVIEW

**Review by EternalRat — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -18,131 +18,137 @@ if (EMSCRIPTEN)
 endif ()
 
 set(SOURCES
-	sources/Models/GameState.hpp
```
Comment (AnonymusRaccoon) [inline CMakeLists.txt:21]: Don't change indentation.

Comment (Octopus773) [reply CMakeLists.txt]: done

File: sources/System/BombHolder/BombHolderSystem.cpp [hunk 2]
```diff
@@ -11,8 +11,14 @@
...
 #include <functional>
 #include <Map/Map.hpp>
+#include "Component/Shaders/Items/BombExplosionShaderComponent.hpp"
+#include <chrono>
+#include "Component/Shaders/ShaderComponent.hpp"
 #include "Component/Collision/CollisionComponent.hpp"
 #include "Component/Tag/TagComponent.hpp"
+#include "Component/Shaders/Items/WhiteShaderComponent.hpp"
+
+float glob = 0;
```
[the remark this answers is not in the document]
Comment (Octopus773) [reply sources/System/BombHolder/BombHolderSystem.cpp:21]: oupsi

File: sources/System/BombHolder/BombHolderSystem.cpp [hunk 3]
```diff
@@ -44,14 +50,40 @@ namespace BBM
...
+				ctx.frameCounter += 0.2;
+				shader.shader.setShaderUniformVar("frame", ctx.frameCounter);
+				shader.shader.setShaderUniformVar("alpha", ctx.alpha);
+				shader.shader.setShaderUniformVar("radius", ctx.explosionRadius);
+			})
 			.addComponent<TimerComponent>(500ms, [](WAL::Entity &explosion, WAL::Wal &wal) {
 				explosion.scheduleDeletion();
 			})
-			.addComponent<Drawable3DComponent, RAY3D::Model>("assets/bombs/explosion/explosion.glb", false,
+			.addComponent<Drawable3DComponent, RAY3D::Model>(GenMeshSphere(0.5, 16, 16),
```
Comment (AnonymusRaccoon) [inline sources/System/BombHolder/BombHolderSystem.cpp:77]: This should be encapsulated.

**Review by AnonymusRaccoon — COMMENTED** [review]
Instead of putting false on every drawable constructor, you could have made a constructor that do not take this boolean and set it to false.

File: sources/System/BombHolder/BombHolderSystem.cpp (hunk 3 above)
Comment (Octopus773) [reply sources/System/BombHolder/BombHolderSystem.cpp]: done

**Pull request merged**